File: ch5/clip8/boj11053/main.py
n = int(input())
arr = list(map(int, input().split()))
d = [1] * n
max_length = 1

for i in range(n):
    for j in range(i):
        if arr[j] < arr[i]:
            if d[j] + 1 < d[i]:
                continue
            d[i] = d[j] + 1
            max_length = max(max_length, d[i])

print(max_length)

File: ch5/clip8/boj11053/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n + 1];
        int d[] = new int[n + 1];

        for(int i = 1; i <= n; i++) {
            arr[i] = sc.nextInt();
        }
        int max = 1;
        for(int i = 1; i <= n; i++) {
            d[i] = 1;
            for(int j = 1; j < i; j++) {
                if(arr[j] < arr[i]) {
                    if(d[j] + 1 < d[i]) continue;
                    d[i] = d[j] + 1;
                    max = Math.max(max, d[i]);
                }
            }
        }
        System.out.println(max);
    }
}
